test/core/security/credentials_test.cc

```diff
@@ -3878,6 +3878,23 @@ TEST(CredentialsTest, TestCompositeChannelCredsCompareSuccess) {
   grpc_channel_credentials_release(composite_creds_2);
 }
 
+TEST(CredentialsTest, RecursiveCompositeCredsDuplicateWithoutCallCreds) {
+  auto* insecure_creds = grpc_insecure_credentials_create();
+  auto inner_fake_creds = MakeRefCounted<fake_call_creds>();
+  auto outer_fake_creds = MakeRefCounted<fake_call_creds>();
+  auto* inner_composite_creds = grpc_composite_channel_credentials_create(
+      insecure_creds, inner_fake_creds.get(), nullptr);
+  auto* outer_composite_creds = grpc_composite_channel_credentials_create(
+      inner_composite_creds, outer_fake_creds.get(), nullptr);
+  auto duplicate_without_call_creds =
+      outer_composite_creds->duplicate_without_call_credentials();
+  auto type = duplicate_without_call_creds->type();
+  ASSERT_NE(type, inner_composite_creds->type());
```

Comment:
Why are we checking that the type does *not* equal something, rather than checking that the type is what we actually expect here?  And actually, why are we checking the type at all rather than just checking that we get the right channel creds instance back?

```
EXPECT_EQ(insecure_creds, duplicate_without_call_creds.get());
```

Note that this should be EXPECT instead of ASSERT.

Reply:
All good points, I was so focused on making sure we _didn't_ return a composite creds I overlooked just checking that we returned exactly the right thing.
Addressed